sklearn/datasets/_samples_generator.py

```diff
@@ -1581,28 +1581,36 @@ def make_spd_matrix(n_dim, *, random_state=None):
             None,
         ],
         "random_state": ["random_state"],
+        "dim": [
+            Interval(Integral, 1, None, closed="left"),
+            Hidden(StrOptions({"deprecated"})),
+        ],
     },
     prefer_skip_nested_validation=True,
 )
 def make_sparse_spd_matrix(
-    dim=1,
+    n_dim=1,
     *,
     alpha=0.95,
     norm_diag=False,
     smallest_coef=0.1,
     largest_coef=0.9,
     sparse_format=None,
     random_state=None,
+    dim="deprecated",
 ):
     """Generate a sparse symmetric definite positive matrix.
 
     Read more in the :ref:`User Guide <sample_generators>`.
 
     Parameters
     ----------
-    dim : int, default=1
+    n_dim : int, default=1
         The size of the random matrix to generate.
 
+        .. versionchanged:: 1.3
+            Renamed from ``dim`` to ``n_dim``.
```

Comment:
You are going to need to have the `dim` params to be documented, otherwise some tests will fail.

You can add it at the end of the Parameters list and add the `.. deprecated:: 1.4` and announce it will be removed in 1.6

Reply:
Ah okay done.